framework/src/testsuite/TestDb.ts

```diff
@@ -0,0 +1,102 @@
+import { existsSync, mkdirSync, readFileSync, unlinkSync, writeFileSync } from 'fs';
+import { resolve as resolvePaths, join as joinPaths } from 'path';
+import { DbItem, DbPlugin, DbPluginConfig, Jovo, JovoResponse } from '..';
+import { HandleRequest } from '../HandleRequest';
+
+export interface FileDbConfig extends DbPluginConfig {
+  directory: string;
+  deleteOnSessionEnded?: boolean;
+}
+
+export class TestDb extends DbPlugin<FileDbConfig> {
+  get dbDirectory(): string {
+    return resolvePaths(process.cwd(), 'dist', this.config.directory);
+  }
+
+  getDefaultConfig(): FileDbConfig {
+    return {
+      ...super.getDefaultConfig(),
+      dbDirectory: '../db/tests',
+    };
+  }
+
+  async mount(handleRequest: HandleRequest): Promise<void> {
+    handleRequest.middlewareCollection.use('request.end', this.loadData.bind(this));
+    handleRequest.middlewareCollection.use('response.start', this.saveData.bind(this));
+    handleRequest.middlewareCollection.use('after.response.end', this.clearData.bind(this));
+  }
+
+  initialize(): void {
+    if (!existsSync(this.dbDirectory)) {
+      mkdirSync(this.dbDirectory, { recursive: true });
+    }
+  }
+
+  getDbItems(primaryKey: string): DbItem[] {
+    if (!existsSync(this.getDbFilePath(primaryKey))) {
+      return [];
+    }
+
+    const rawDbData: string = readFileSync(this.getDbFilePath(primaryKey), 'utf-8');
+    const dbItems: DbItem[] = rawDbData.length ? JSON.parse(rawDbData) : [];
+    return dbItems;
+  }
+
+  getDbItem(primaryKey: string): DbItem | undefined {
+    const dbItems: DbItem[] = this.getDbItems(primaryKey);
+    return dbItems.find((dbItem: DbItem) => dbItem.id === primaryKey);
+  }
+
+  async loadData(jovo: Jovo): Promise<void> {
```

Comment:
Needs to be updated after changes in the DbPlugin: https://github.com/jovotech/jovo-framework/pull/1031